pages/myPlants.js

```diff
@@ -0,0 +1,45 @@
+import Footer from "@/components/Footer";
+import PlantList from "@/components/PlantList";
+import styled from "styled-components";
+import useLocalStorageState from "use-local-storage-state";
+
+export default function MyPlantsPage() {
+  const [plants, setPlants] = useLocalStorageState("favourites");
+
+  const favouritePlants = plants
+    ? plants.filter((plant) => plant.isFavourite)
+    : [];
+
+  function toggleFavourite(id) {
+    setPlants((prevPlants) =>
+      prevPlants.map((plant) =>
+        plant.id === id ? { ...plant, isFavourite: !plant.isFavourite } : plant
+      )
+    );
+  }
```

Comment:
💡 Moving the state from `index.js`  to `_app.js` is a better solution for shared state management across pages, as it allows the plants `data` and the `toggleFavourite `function to be accessible in multiple components without duplicating state logic. 

Please have a look into the Handout from the [global state](https://github.com/neuefische/hh-web-24-8/blob/main/sessions/react-global-state/react-global-state.md) session 

**The advantages:**
- With plants in _app.js, all pages share the same data, ensuring consistency.
- Each page can focus on displaying data without managing its own local storage or duplicate state.
- As the app grows, maintaining centralised state will make it easier to add more features without duplication or synchronization issues.